SmartDeviceLinkTests/ProxySpecs/SDLSecondaryTransportManagerSpec.m

```diff
@@ -483,36 +488,62 @@ + (void)swapConnectionMethods {
         });
     });
 
-
     describe(@"In Configured state", ^{
         describe(@"if secondary transport is iAP", ^{
             beforeEach(^{
                 // in this case we assume the primary transport is TCP
                 testPrimaryProtocol = [[SDLProtocol alloc] init];
                 testPrimaryTransport = [[SDLTCPTransport alloc] init];
                 testPrimaryProtocol.transport = testPrimaryTransport;
+
                 dispatch_sync(testStateMachineQueue, ^{
                     [manager startWithPrimaryProtocol:testPrimaryProtocol];
                 });
 
                 manager.secondaryTransportType = SDLTransportSelectionIAP;
             });
 
-            it(@"should transition to Connecting state", ^{
-                // setToState cannot be used here, as the method will set the state after calling didEnterStateConfigured
-                dispatch_sync(testStateMachineQueue, ^{
-                    [manager.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+            context(@"before the security manager is set by register app interface response", ^{
+                it(@"should transition to Connecting", ^{
```

Comment:
It should transition to connecting before the security manager is set or isAppReady is set? That doesn't seem right?

Reply:
i fixed the code to match this catch! the manager would connect on an or case